### PR TITLE
updated the css file and js

### PR DESCRIPTION
**Pull request by dosanjhkiran**
please review the code

### CODE REVIEW

File: assets/js/dashboard.js [hunk 1]
```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
...
-            // Manage data here
+            let daysData = get5DayForecast(data.list);
+            display5DayForecast(daysData);
         })
         .catch(error => {
             console.log(error);
         });
 
     // API Calls to get location details
-    // API fetch request | "https://www.googleapis.com/customsearch/v1?key={API key}&cx={PSE ID}&q={city name}&searchType=image&num=1"
```
Comment (ANAZ0004) [inline assets/js/dashboard.js:89]: Why trim this documentation?

File: assets/js/dashboard.js [hunk 2]
```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
...
             console.log(data); // SEE: data format.
-            locationDesc.innerHTML = data.items[0].htmlSnippet.concat(" <a href=", data.items[0].formattedUrl, ">Read More</a><br>Result retrieved from ", data.items[0].displayLink)/* City info from API */;
+            locationDesc.innerHTML = data.items[0].htmlSnippet.concat(" <a href=", data.items[0].formattedUrl, ">Read More</a><br>Result retrieved from ", data.items[0].displayLink);/* City info from API */;
         })
         .catch(error => {
-            console.log(error);
+            console.log("Error:", error); // prints the error in the console
         });
 
-    // API fetch request | "https://www.googleapis.com/customsearch/v1?key={API key}&cx={PSE ID}&q={city name}&searchType=image&num=1"
```
Comment (ANAZ0004) [inline assets/js/dashboard.js:107]: Why trim this documentation?

File: assets/js/dashboard.js [hunk 3]
```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
...
-    // This is for a Google SafeSearch.
-    let fetchURLGoogleDesc = googleAPIURL.concat("key=", googleAPIKeyAN, "&cx=", googlePSEID, "&q=", searchQuery, "&num=1")
-    fetch(fetchURLGoogleDesc)
-        .then(response => {
-            if (!response.ok) {
-                throw new Error("API response was not okay. (Google query)");
-            }
-            return response.json();
-        })
+    fetchWithRetry(googleAPIURL.concat("key=", googleAPIKeyAN, "&cx=", googlePSEID, "&q=", searchQuery, "&num=1"))
```
Comment (ANAZ0004) [inline assets/js/dashboard.js]: fetchWithRetry is not an existing function, **we would need to make our own.** (May be outside the course content).
I think it is better to keep the old way but **it's okay to remove declaration of fetchURLGoogleDesc variable** and instead place the concatenation inside the function parameter.
**This currently does not work.**
"Uncaught ReferenceError: fetchWithRetry is not defined."

File: assets/js/dashboard.js [hunk 4]
```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
...
-    // This is for a Google Images SafeSearch.
-    let fetchURLGoogleImage = googleAPIURL.concat("key=", googleAPIKeyAN, "&cx=", googlePSEID, "&q=", searchQuery, "&searchType=image&num=1")
-    fetch(fetchURLGoogleImage)
-        .then(response => {
-            if (!response.ok) {
-                throw new Error("API response was not okay. (Google images query)");
-            }
-            return response.json();
-        })
+    fetchWithRetry(googleAPIURL.concat("key=", googleAPIKeyAN, "&cx=", googlePSEID, "&q=", searchQuery, "&searchType=image&num=1"))
```
Comment (ANAZ0004) [inline assets/js/dashboard.js]: Same as above.
fetchWithRetry is not an existing function, **we would need to make our own.** (May be outside the course content).
I think it is better to keep the old way but **it's okay to remove declaration of fetchURLGoogleImage variable** and instead place the concatenation inside the function parameter.
**This currently does not work.**
"Uncaught ReferenceError: fetchWithRetry is not defined."

File: assets/css/style.css
Comment (ANAZ0004) [inline assets/css/style.css:1]: CSS is great and functional!

Edited:
Small note regarding footer behaviour below.

File: assets/js/dashboard.js [hunk 5]
```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
...
+            });
+        }
+    });
+
+    return daysData.slice(0, 5); // Ensure only 5 days are returned
+}
+
+// Function to display 5-day forecast data
+function display5DayForecast(daysData) {
+    let forecastContainer = document.getElementById("forecast-container");
```
Comment (ANAZ0004) [inline assets/js/dashboard.js:145]: There is currently no element with ID "forecast-container", thus, this function does not work.

File: assets/js/dashboard.js [hunk 6]
```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
...
+        forecastDiv.innerHTML = `
+            <h3>${dayData.date}</h3>
+            <p>Temp: ${temperatureToCelcius(dayData.temp)}</p>
+            <p>Weather: ${dayData.weather}</p>
+        `;
+        forecastContainer.appendChild(forecastDiv);
+    });
+}
+// Function to display current wind speed
+function displayWindSpeed(speed, elementId) {
```
Comment (ANAZ0004) [inline assets/js/dashboard.js:160]: Great work with this function, however, it is not being called anywhere at the moment, thus, is not being executed. Feel free to implement code to address this.

**Review by ANAZ0004 — CHANGES_REQUESTED** [review]
Great work with the CSS, the website is functional and stylish!
Please take a look at some comments and suggestions I have attached to the `styles.css` and `dashboard.js` files. I have identified a few issues and suggested ways to address them.
Great job, Kirandeep!
Preview of reviewed build:
image

File: assets/css/style.css [hunk 7]
```diff
@@ -1,54 +1,147 @@
...
+    margin: 5px 0;
+}
+
+/* Footer styling */
+footer {
+    width: 100%;
+    background-color: var(--light-grey-bg-color);
+    text-align: center;
+    padding: 10px 0;
+    position: absolute;
```
Comment (ANAZ0004) [inline assets/css/style.css]:
```suggestion
    position: fixed;
```
For correct footer behaviour :)

File: assets/css/style.css [hunk 8]
```diff
@@ -1,54 +1,147 @@
...
+    bottom: 0;
```
Comment (ANAZ0004) [inline assets/css/style.css]:
```suggestion
    left: 0;
    bottom: 0;
```
To get rid of small gap present in certain viewports.

**Review by ANAZ0004 — CHANGES_REQUESTED** [review]
Just a couple CSS changes regarding footer behaviour.

**Review by jairatagle — APPROVED** [review]

**Review by ANAZ0004 — APPROVED** [review]

**Review by jairatagle — APPROVED** [review]

**Review by jairatagle — APPROVED** [review]

**Review by dosanjhkiran — COMMENTED** [review]
goodjob

**Pull request merged**